packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/CodeWriterTests.cs

```diff
@@ -264,5 +266,496 @@ public void TestWriteMethodDeclarationNoScope_ConstructorSignature()
+            var p2 = new Parameter("key", $"key", typeof(string), null, ValidationType.None, null);
+            var indexer2 = new IndexerDeclaration($"To test an expression property with int type", MethodSignatureModifiers.Public, typeof(int), p2, new ExpressionPropertyBody(Literal(299792458)), ExplicitInterface: typeof(IReadOnlyDictionary<string, int>));
+
+            var codeWriter = new CodeWriter();
+            codeWriter.WriteProperty(indexer1);
+            codeWriter.Append($"// test comment");
+            codeWriter.WriteProperty(indexer2);
+
+            var expected = new StringBuilder()
+                .Append(_header)
+                .Append("using System.Collections.Generic;").Append(NewLine).Append(NewLine)
+                .Append("/// <summary> To test an expression property with string type. </summary>").Append(NewLine)
+                .Append("public string global::System.Collections.Generic.IReadOnlyList<string>.this[int index] => \"abc\";").Append(NewLine)
+                .Append("// test comment").Append(NewLine)
+                .Append("/// <summary> To test an expression property with int type. </summary>").Append(NewLine)
+                .Append("public int global::System.Collections.Generic.IReadOnlyDictionary<string, int>.this[string key] => 299792458;").Append(NewLine)
+                .ToString();
+
+            var result = codeWriter.ToString();
+
+            Assert.AreEqual(expected, result);
+        }
+
+        [Test]
+        public void CodeWriter_WriteProperty_IndexerProperty_MethodPropertyBody()
+        {
+            var p1 = new Parameter("p1", $"p1", typeof(int), null, ValidationType.None, null);
+            var indexer1 = new IndexerDeclaration($"To test a method property without a setter", MethodSignatureModifiers.Public, typeof(string), p1, new MethodPropertyBody(Return(Literal("abc"))));
+            var p2 = new Parameter("p2", $"p2", typeof(int), null, ValidationType.None, null);
+            var indexer2 = new IndexerDeclaration($"To test a method property with a setter", MethodSignatureModifiers.Public, typeof(string), p2, new MethodPropertyBody(Return(Literal("abc")), Assign(new FormattableStringToExpression($"Property2"), new KeywordExpression("value", null))));
+            var p3 = new Parameter("p3", $"p3", typeof(int), null, ValidationType.None, null);
+            var indexer3 = new IndexerDeclaration($"To test a method property with an internal setter", MethodSignatureModifiers.Public, typeof(string), p3, new MethodPropertyBody(Return(Literal("abc")), Assign(new FormattableStringToExpression($"Property3"), new KeywordExpression("value", null)), MethodSignatureModifiers.Internal));
+
+            var codeWriter = new CodeWriter();
+            codeWriter.WriteProperty(indexer1);
+            codeWriter.Append($"// test comment");
+            codeWriter.WriteProperty(indexer2);
+            codeWriter.WriteProperty(indexer3);
+
+            var expected = new StringBuilder()
```

Comment:
do we think its better to have asset files as our expected values here?

Reply:
Sure I could create one for that

Reply:
well sorry I did not manage to do it because of some CRLF/LF issues :(
also i notice other test cases in this file is also using StringBuilders therefore I followed.
Maybe we could do that later?

Reply:
created an issue: https://github.com/microsoft/typespec/issues/3311